config/passport.js

```diff
@@ -10,14 +10,29 @@ passport.use(
             passwordField: "user[password]"
         },
         function(email, password, done) {
-            User.findOne({ email: email })
+            User
+            .findOne({
+                where: {
+                    email: email
+                }
+            })
                 .then(function(user) {
-                    if (!user || !user.validPassword(password)) {
+
+                   const hash = crypto
+                   .pbkdf2Sync(password,user.salt, 10000, 512, "sha512")
+                   .toString("hex");
+
+                   
+                    if (!user) {
                         return done(null, false, {
-                            errors: { "email or password": "is invalid" }
+                            errors: { "email": "is invalid" }
+                        });
+                    }
+                    if (!(hash ===  user.hash)) {
+                        return done(null, false, {
+                            errors: { "email": "is invalid" }
```

Comment:
The error message should be `{ "password": "is invalid" }` for this logic branch.